src/domain/users-ips/ip-metadata-validator.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  MissingIPMetadataError,
+  InvalidIPMetadataCountryError,
+  InvalidIPMetadataProxyError,
+  InvalidIPMetadataASNError,
+} from "@domain/errors"
+
+export const IPMetadataValidator = ({
+  denyIPCountries,
+  allowIPCountries,
+  denyASNs,
+  allowASNs,
+}: IPMetadataValidatorArgs): IPMetadataValidator => {
+  const validateForReward = (ipMetadata?: IPType): true | ApplicationError => {
```

Comment:
Same here dependency graph is app -> domain. Not the other way around. Domain should return specific errors.

Reply:
Fixed and `PhoneMetadataValidator` too https://github.com/GaloyMoney/galoy/pull/1232/commits/9333231b2b3e87f0f2eb04110a8f03bcee7ac70b